app/code/core/Mage/Core/Model/Resource/Store/Collection.php

```diff
@@ -193,14 +189,6 @@ public function initConfigCache()
         $data = [];
         $needsRefresh = false;
 
-        if ($globalConfigCache !== false) {
-            try {
-                $data = unserialize($globalConfigCache);
-            } catch (Exception $exception) {
-                Mage::logException($exception);
-            }
-        }
-
```

Comment:
the old version was storing only strings in the cache, this new version stores everything

Reply:
But since we don't need to `unserialize`, it would just be:

```php
        if ($globalConfigCache !== false) {
            $data = $globalConfigCache;
        }
```

And then also removing the `serialize` call at the end of this method.

Reply:
@fballiano also tagging in case you won't see this otherwise

Reply:
@justinbeaty you got a big bug with that code removal, which prevented the needsRefresh from working correctly!

I've committed https://github.com/MahoCommerce/maho/pull/165/commits/d3a661eb6592b4fbd348e683f0f07b839eb7d5ac which also removes `initConfigCache()` from `app/code/core/Mage/Core/Model/Store.php` since it seems not to be used anywhere and phpstan is not complaining nor can I see any error.

🙏

Reply:
I had written in my notes to double check this, but almost forgot before finalizing my review. Glad we got it fixed. :)

I agree with the removal of that old function. I think it was just legacy stuff since it saved a separate cache entry for each store config.